fs_mark/fs_mark.py

```diff
@@ -0,0 +1,197 @@
+            line_dict['closeAvg'] = splited_line[18]
+            line_dict['closeMax'] = splited_line[19]
+
+            # If files are kept, fs_mark will unlink them so unlink time can be caught
+            if not ('-k' in test_name or '-L' in test_name or '-F' in test_name):
+                line_dict['unlinkMin'] = splited_line[20]
+                line_dict['unlinkAvg'] = splited_line[21]
+                line_dict['unlinkMax'] = splited_line[22]
+
+            # Write timings to file if exists
+            try:
+                with open(os.path.expanduser(logs_path), 'a') as file:
+                    if first_loop is True:
+                        first_loop = False
+                        file.write(test_name + ' ' + str(datetime.now()) + '\n')
+                    else:
+                        file.write('\n')
+                    yaml.dump(line_dict, file)
+
+            except FileNotFoundError:
+                pass
+
+            subresult = TestSubResult(name=str(test_name), status=Status.OK)
+
+            for name, value in line_dict.items():
+                if not t_data.timing_dict[(target, name)][0] <= int(value) <= t_data.timing_dict[(target, name)][1]:
+                    test_status = Status.FAIL
+                    subresult.status = Status.FAIL
+                    subresult.msg += "\n\t" + name + " exec time - " + value + \
+                                     " out of range [" + str(t_data.timing_dict[(target, name)][0]) + \
+                                     " - " + str(t_data.timing_dict[(target, name)][1]) + "]"
+
+            if subresult.status == Status.FAIL:
+                subresult.msg += "\n\n\tF_Use%: " + str(f_use)
+                subresult.msg += "\n\tCount: " + str(count)
+                subresult.msg += "\n\tSize: " + str(size)
+                subresult.msg += "\n\tFiles/sec: " + str(files_sec)
+                subresult.msg += "\n\tApp overhead: " + str(app_overhead) + "\n\t"
+                result.add_subresult_obj(subresult)
+                break
+
+        elif idx == 2:
+            if got_time is False:
+                test_status = Status.FAIL
+                subresult = TestSubResult(name=str(test_name), status=Status.FAIL)
+                subresult.msg += "\n\tInsufficient free space"
+                result.add_subresult_obj(subresult)
+            break
+        elif idx == 3:
+            end_time = time.time()
+            if got_time is False:
+                test_status = Status.FAIL
+                subresult = TestSubResult(name=str(test_name), status=Status.FAIL)
+                subresult.msg += "\n\tGot no timings"
+            result.add_subresult_obj(subresult)
+            break
+        elif idx == 4:
+            test_status = Status.FAIL
+            subresult = TestSubResult(name=str(test_name), status=Status.FAIL)
+            subresult.msg += "\n\t" + parsed['err']
+            result.add_subresult_obj(subresult)
+            break
```

Comment:
I would try to simplify it, maybe something like that (not tested):
```suggestion
	if idx in [2, 3, 4]:
		if idx == 2 and not got_time:
			msg= "\n\tInsufficient free space"
		elif idx == 3:
			end_time = time.time()
			if not got_time:
				msg = "\n\tGot no timings"
		elif idx == 4:
			msg = "\n\t" + parsed['err']
		
		test_status = Status.FAIL
		subresult = result.add_subresult(subname=str(test_name, status=Status.FAIL)
		subresult.msg += msg
```